dotnet/src/Connectors/Connectors.Google/GeminiPromptExecutionSettings.cs

```diff
@@ -262,6 +263,32 @@ public string? CachedContent
         }
     }
 
+    /// <summary>
+    /// Configuration for the thinking budget in Gemini 2.5.
+    /// **This property is specific to Gemini 2.5 and similar experimental models.**
+    /// </summary>
+    [JsonPropertyName("thinking_config")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public GeminiThinkingConfig? ThinkingConfig
+    {
+        get => this._thinkingConfig;
+        set
+        {
+            this.ThrowIfFrozen();
+            bool isGemini25 = false;
+            if (this.ModelId != null)
+            {
+                isGemini25 = this.ModelId?.StartsWith("gemini-2.5", StringComparison.OrdinalIgnoreCase) ?? false;
+            }
+
+            if (!isGemini25 && value != null)
+            {
+                throw new InvalidOperationException("ThinkingConfig is only applicable to Gemini-2.5 models.");
+            }
+            this._thinkingConfig = value;
```

Comment:
Remove any conditional related to the model, we should not validate this on our side, leave it to the server giving back the expected 400.

> [!NOTE]
> Side note here:
> As a good design practice we suggest avoiding as much as possible throwing exceptions from Properties Getters / Setters.

```suggestion
            this._thinkingConfig = value;
```